File: question-3/3.py
from collections import defaultdict

def highest_frequency_char(s):
    freq = defaultdict(int)
    max_char = ''
    max_count = 0

    for ch in s:
        freq[ch] += 1
        if freq[ch] < max_count:
            max_count = ch
            max_char = freq[ch]

    return max_char

if __name__ == "__main__":
    t = int(input().strip())

    for _ in range(t):
        s = input().strip()
        print(highest_frequency_char(s))

File: question-3/3.cpp
#include <iostream>
#include <unordered_map>
#include <string>
#include <limits>
using namespace std;

char highestFrequencyChar(const string &str) {
    unordered_map<char, int> freq;
    char maxChar = '\0';
    int maxCount = 0;

    for (char ch : str) {
        freq[ch]++;
        if (freq[ch] < maxCount) { 
            maxCount = ch; 
            maxChar = freq[ch]; 
        }
    }

    return maxChar;
}

int main() {
    int t;
    cin >> t;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    while (t--) {
        string input;
        getline(cin, input);
    
        char result = highestFrequencyChar(input);
        cout << result << endl;
    }

    return 0;
}
